### ISSUE TITLE
Further Array#any? optimization

### ISSUE THREAD

**Issue by elia** [opal/opal#1517]
Update `Array#any?` implementation to:

``` rb
`if (self.length === 0) return false;`
super
```

see discussion: https://github.com/opal/opal/pull/1514#issuecomment-231037728

**Comment by karimmtarek** [comment]
@elia I think this is already done [there](https://github.com/opal/opal/blob/master/opal/corelib/array.rb#L443), no?

**Comment by elia** [comment]
@karimmtarek you're right, the difference in the version proposed in this issue is that avoids a number of methods calls and the Ruby `if` logic that is more complex. So it should yield some perf improvements

**Comment by karimmtarek** [comment]
@elia so what you are looking for(as you already proposed [here](https://github.com/opal/opal/pull/1514#issuecomment-231037728)) is this:

```
`if (self.length === 0) return false;`
super
```

Right?

**Comment by elia** [comment]
@karimmtarek yes that's right

**Comment by elia** [comment]
solved by @CicholGricenchos in #1667
